src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.unit.netcoreapp.cs

```diff
@@ -0,0 +1,520 @@
+                string dirNameSecond = Path.Combine(testDirectory.Path, "dirfoo");
+                string[] expectedPaths = new string[] { dirName, dirNameSecond };
+
+                watcher.Filters.Add(Path.GetFileName(dirName));
+                watcher.Filters.Add(Path.GetFileName(dirNameSecond));
+
+                Action action = () => Directory.Delete(dirName);
+                Action cleanup = () => Directory.CreateDirectory(dirName);
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
+
+                action = () => Directory.Delete(dirNameSecond);
+                cleanup = () => Directory.CreateDirectory(dirNameSecond);
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Create()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            using (var watcher = new FileSystemWatcher(testDirectory.Path))
+            {
+                string fileName = Path.Combine(testDirectory.Path, "file");
+                string secondFileName = Path.Combine(testDirectory.Path, "Secondfile");
+                string[] expectedPaths = new string[] { fileName, secondFileName };
+
+                watcher.Filters.Add(Path.GetFileName(fileName));
+                watcher.Filters.Add(Path.GetFileName(secondFileName));
+
+                Action action = () => File.Create(fileName).Dispose();
+                Action cleanup = () => File.Delete(fileName);
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, action, cleanup, expectedPaths);
+
+                action = () => File.Create(secondFileName).Dispose();
+                cleanup = () => File.Delete(secondFileName);
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Created, action, cleanup, expectedPaths);
+            }
+        }
+
+        [Fact]
+        public void FileSystemWatcher_File_Delete()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            using (var watcher = new FileSystemWatcher(testDirectory.Path))
+            {
+                string fileName = Path.Combine(testDirectory.Path, "file");
+                string secondFileName = Path.Combine(testDirectory.Path, "Secondfile");
+                string[] expectedPaths = new string[] { fileName, secondFileName };
+
+                watcher.Filters.Add(Path.GetFileName(fileName));
+                watcher.Filters.Add(Path.GetFileName(secondFileName));
+
+                Action action = () => File.Delete(fileName);
+                Action cleanup = () => File.Create(fileName).Dispose();
+                cleanup();
+
+                ExpectEvent(watcher, WatcherChangeTypes.Deleted, action, cleanup, expectedPaths);
```

Comment:
> Why pass both paths when you're only expecting one?

For some reason, in the second call to the ```ExpectEvent``` , expectedPath[0] is picking the first value only i.e. path to ```file``` rather than path to ```secondFile```.
I tried debugging it. The only statement in the second call to ```ExpectEvent``` where the ```expectedPath[0]``` is not ```secondFile```  suddenly is https://github.com/dotnet/corefx/blob/master/src/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs#L84